skimage/filters/_fft_based.py

```diff
@@ -0,0 +1,144 @@
+        FFT.
+    high_pass : bool, optional
+        Whether to perform a high pass filter. If False, a low pass filter is
+        performed.
+    order : float, optional
+        Order of the filter which affects the slope near the cut-off. Higher
+        order means steeper slope in frequency space.
+    channel_axis : int, optional
+        If there is a channel dimension, provide the index here. If None
+        (default) then all axes are assumed to be spatial dimensions.
+    preserve_range : bool, optional
+        Whether to keep the original range of values. Defaults to False. If
+        False, image is returned with dtype float with values scaled between 0
+        and 1. If a complex-valued image is passed, no rescaling is performed.
+
+    Returns
+    -------
+    result : ndarray
+        The Butterworth-filtered image.
+
+    Notes
+    -----
+    A band-pass filter can be achieved by combining a high pass and low
+    pass filter.
+
+    The literature contains multiple conventions for the functional form of
+    the Butterworth filter. Here it is implemented as the n-dimensional form of
+
+    .. math::
+        \\frac{1}{1 - \\left(\\frac{f}{c*f_{max}}\\right)^{2*n}}
+
+    with :math:`f` the absolute value of the spatial frequency, :math:`c` the
+    ``cutoff_frequency_ratio`` and :math:`n` the ``order`` modeled after [2]_
+
+    Examples
+    --------
+    Apply a high pass and low pass Butterworth filter to a grayscale and
+    color image respectively:
+
+    >>> from skimage.data import camera, astronaut
+    >>> from skimage.filters import butterworth
+    >>> high_pass = butterworth(camera(), 0.07, True, 8)
+    >>> low_pass = butterworth(astronaut(), 0.01, False, 4, channel_axis=-1)
+
+    References
+    ----------
+    .. [1] Butterworth, Stephen. "On the theory of filter amplifiers."
+           Wireless Engineer 7.6 (1930): 536-541.
+    .. [2] Russ, John C., et al. "The image processing handbook."
+           Computers in Physics 8.2 (1994): 177-178.
+
+    """
+    fft_shape = (image.shape if channel_axis is None
+                 else np.delete(image.shape, channel_axis))
+    is_real = np.isrealobj(image)
+    wfilt = _get_ND_butterworth_filter(
+        fft_shape, cutoff_frequency_ratio, order, high_pass, is_real
+    )
+    axes = np.arange(image.ndim)
+    if channel_axis is not None:
+        axes = np.delete(axes, channel_axis)
+        abs_channel = channel_axis % image.ndim
+        post = image.ndim - abs_channel - 1
+        sl = ((slice(None),) * abs_channel + (np.newaxis,) +
+              (slice(None),) * post)
+        wfilt = wfilt[sl]
+    if is_real:
+        butterfilt = fft.irfftn(wfilt * fft.rfftn(image, axes=axes),
+                                s=fft_shape, axes=axes)
+        out_range = dtype_limits(image) if preserve_range else (0.0, 1.0)
+        butterfilt = rescale_intensity(butterfilt, out_range=out_range)
```

Comment:
```suggestion
        if not preserve_range:
            image = img_as_float(image)        
        butterfilt = fft.irfftn(wfilt * fft.rfftn(image, axes=axes),
                                s=fft_shape, axes=axes)
```

I would have instead, expected behavior like the above for `preserve_range`. The current code will rescale the range for pretty much all inputs even when preserve_range is True. For example, floating point inputs will not have their range preserved.

@jni, thoughts? If we change this here we will need to change the data files used in the tests.

Reply:
Gah. I think you're right @grlee77. 🤦

Reply:
sorry for replying so late, didn't have time for a couple days. I think I implemented it in this way because the fft + ifft may introduce image intensities outside the original data-type range. For example if you put in a uint8 image, you may get negative intensities. This is because when you remove some spatial frequencies, you might get "jumps" in places where there are edges. My original interpretation of `preserve_range` was that the output image should fit in the same dtype range as the input image so that it could also be cast as the same dtype. This is why I do the rescaling after the filter. But I now see your point that this has the potentially unwanted side effect of stretching the histogram to the extrema, so probably your solution is better. Note however that in your implementation, it is still possible to get an image that is not rescaled to a range [0, 1] or [-1, 1] if `preserve_range=False`, but I don't see another appropriate alternative that won't stretch the histogram. I suggest the following:

* I change the implementation to what @grlee77 suggests
* I add some clarification in the `Notes` section about `preserve_range` that it is possible to still get some outlier intensities
* I remove the test data files and hardcode new test data into the test files.

Alternatively, since the option doesn't seem to make much sense anyway I remove the `preserve_range` argument. Any vote on this?